### ISSUE TITLE
How to build my own dataset?

### ISSUE THREAD

**Issue by zixib1108** [koyeongmin/PINet_new#31]

**Comment by zixib1108** [comment]
Also, after running train.py from my own dataset, There is a FileNotFoundError which links to the path: /home/kym/Dropbox/eval_result2_3_.txt
How should I modify this part?

**Comment by zixib1108** [comment]
Still, I built my own dataset, and train.py works. However, when generating results, it shows that ("We do not get the predictions of all the test tasks"). Any suggestions on how to fix this problem? 
Best!

**Comment by koyeongmin** [comment]
" FileNotFoundError which links to the path: /home/kym/Dropbox/eval_result2_3_.txt" is for saving results.
You need to change savefile path to your system.
And I don't know where the error, ("We do not get the predictions of all the test tasks"), occurs.
Could you tell me more detail?
Thank you!

**Comment by zixib1108** [comment]
Thats a problem in evalutaion.py a function called bench_one_submit (pred_file, gt_file). In main, the argument are "test_result_0_3.json", "test_label.json"

**Comment by koyeongmin** [comment]
It looks also path problem.
Could you check your test result save path?
When the code makes savefiles, "test_result_0_3.json", "test_label.json" should be in the same directory to source codes.
Thank you!

**Comment by zixib1108** [comment]
Yes, I already check the problem, they do exist in the same directory where train.py locates. The trigger is the if statement which judge whether those two jsons have the same length of the lines. But I don't know why they are not same and I don't how these two files are generated

**Comment by zixib1108** [comment]
BTW, may I know the time that you are available? I am new to this PINet and I might need more help in the future to train my own built datasets. Thanks so much for your great time and help! I am in China right now, so I think I need to post issues in the time zone that are convenient for you.
Best wishes.

**Comment by koyeongmin** [comment]
If you are using your own dataset, could you check the format of test_label.json. and test_result_*_.json generated by your dataset?
Provided test_label.json file in this repository is for TuSimple dataset. 
And, I'm fine any time before dinner.
Thank you!

**Comment by zixib1108** [comment]
I think it is in the way of Tusimple dataset. I will double check this point. Thanks for your help! I will update if there is any issues or progress!
Take care!

**Comment by zixib1108** [comment]
HI, when I fix these problems, new error occured. In agent.py, there is a variable called trim which showed undefined. What should the value of trim be? Any suggestions on it?
Best!

**Comment by zixib1108** [comment]
BTW, what should be the ratio of test and train. From the provided Tusimple dataset, I found that the test dataset is larger than training dataset

**Comment by koyeongmin** [comment]
"trim" is just for resuming training. If you are training the model from scratch, you can set trim=0.
There is no fixed ratio of test and train set. In most datasets, the test set is less than the train set.
Thank you!

**Comment by zixib1108** [comment]
Actually, I am using this algo to detect wires (cell phone charge wires, PC/TVs power wires, Power panel wires...) instead of Lanes. Hope this works. I will update if I get some progress.

**Comment by zixib1108** [comment]
![warning](https://user-images.githubusercontent.com/87160436/125743321-08fd4544-9330-44e2-b74a-91ff5fa2c081.jpg)
When I training, this warning occurred, and the evaluation always stucked at 75%

**Comment by zixib1108** [comment]
Meanwhile, when I training from my own models. What would be the value in trainpy line 54: lane_agent.load_weights(<>, "tensor(<>)")

**Comment by koyeongmin** [comment]
Oh, this is my mistake. the progress bar is not stopped at 100% when the evaluation is finished. The length can be changed according to batch size and test set size.

**Comment by zixib1108** [comment]
lol, thanks not a big problem. My model works fine and the detect of those wires are not bad, still training now

**Comment by zixib1108** [comment]
What is the tensor value range for a good model, around 1? or below 1? And how to make those detect dots much more smaller?
Thanks.

**Comment by zixib1108** [comment]
If Just want to detect whether it is a lane or not, instead of detecting which specific lane it is, how should I modify the code, where should I pay attention to when modify the code
thanks!

**Comment by koyeongmin** [comment]
If you want to detect only whether it is a lane or not, you may only need a confidence branch in network. (also only confidence loss function is necessary)

**Comment by zixib1108** [comment]
Thanks!
BTW, How to use test mode 2 to visualize the results?

**Comment by koyeongmin** [comment]
To test mode 2, you need to change image path to your system.
Thank you!

**Comment by zixib1108** [comment]
Thanks, has tested
